ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Serialisation/SerialiseNodeToClient.hs

```diff
@@ -135,6 +139,50 @@ dispatchDecoderErr ccfg version =
 after :: (a -> b -> d -> e) -> (c -> d) -> a -> b -> c -> e
 after f g x y z = f x y (g z)
 
+{-------------------------------------------------------------------------------
+  Ledger Config
+-------------------------------------------------------------------------------}
+
+instance All SerialiseConstraintsHFC xs
+  => SerialiseNodeToClient (HardForkBlock xs) (PerEraLedgerConfig xs) where
+  encodeNodeToClient (HardForkCodecConfig (PerEraCodecConfig ccfgs)) version (PerEraLedgerConfig xs) = case version of
+    HardForkNodeToClientDisabled versionX -> case xs of
+      x :* _ -> case ccfgs of
+        cfg :* _ -> encodeNodeToClient cfg versionX x
+    HardForkNodeToClientEnabled _ subVersions ->
+      let -- Encoding of the components of xs, filtering out the eras disabled in subVersions.
```

Comment:
Are we filtering because if we negotiated a version which excludes later eras that we're aware of that means they are not even aware of those versions, and so they wouldn't expect them in the serialization?

That makes sense, but it would probably be worth adding a comment on this instance that spells that out for future clarity 👍